Build Rails for EL9

evgeni opened the pull request:
<!--
If your package needs to be released within one or more release streams, and/or distributions, please open PRs to each of those branches respectively. The easiest way to do this is to make the initial commit for the mainline branch (e.g. rpm/develop or deb/develop) and then cherry pick the commit hash onto each subsequent branch.

The Foreman Community supports the `develop` branch for active development and the latest two releases.
You can view the currently supported versions on [theforeman.org](https://theforeman.org/).
 
RPM Example:

    git checkout -b rpm/develop-foreman-tasks-1.0.1 rpm/develop

    # Make changes to update package

    git commit -a -m 'Release foreman-tasks-1.0.1'
    COMMIT=`git rev-parse HEAD`

    git checkout -b rpm/1.20-foreman-tasks-1.0.1-1.20 rpm/1.20
    git cherry-pick -x $COMMIT

DEB Example:

    git checkout -b deb/develop-foreman-tasks-1.0.1 deb/develop

    # Make changes to update package

    git commit -a -m 'Release foreman-tasks-1.0.1'
    COMMIT=`git rev-parse HEAD`

    git checkout -b deb/1.20-foreman-tasks-1.0.1-1.20 deb/1.20
    git cherry-pick -x $COMMIT

See Foreman's [plugin maintainer documentation](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.
-->

ekohl reviewed (commented):
I get the impression our Jenkins job doesn't understand branching: it can't find altered packages since we're not revbumping. I'd prefer it if we could build a new OS without revbumping. Any thoughts on that?

evgeni commented:
> I get the impression our Jenkins job doesn't understand branching: it can't find altered packages since we're not revbumping. I'd prefer it if we could build a new OS without revbumping. Any thoughts on that?

Correct. My intent was to run a local `obal release rails_core_packages` which will detect that they are built on el8 but not el9 and only do el9 builds.

evgeni commented on package_manifest.yaml:
```diff
@@ -65,6 +65,24 @@ copr_projects:
+        - name: "rhel-{{ rhel_9 }}-x86_64"
+          modules: "{{ core_modules }}"
+          external_repos:
+            - "https://yum.puppet.com/puppet7/el/{{ rhel_9 }}/x86_64/"
```
That's a fair point, let me prepare this in another PR and then rebase here.

evgeni replied:
https://github.com/theforeman/foreman-packaging/pull/10018

ehelms commented on package_manifest.yaml:
```diff
@@ -66,6 +66,23 @@ copr_projects:
           comps_file: "{{ inventory_dir }}/comps/comps-foreman-el{{ rhel_8 }}.xml"
           buildroot_packages: "{{ core_buildroot_packages }}"
+    foreman-copr-with-el9:
+      copr_project_name: "foreman-{{ foreman_version }}-staging"
```
If this does not change, why add a new entry?

evgeni replied:
how else would we refer to it when a group is configured to use the `with-el9` config?

evgeni commented:
Create stuff in Copr:

```
% obal copr-project foreman-copr-with-el9

PLAY [Configure Copr projects] ********************************************************************************************************************************************************************************************************

TASK [copr_project : include_tasks] ***************************************************************************************************************************************************************************************************
included: /home/evgeni/Devel/theforeman/obal/obal/data/roles/copr_project/tasks/create.yml for foreman-copr-with-el9

TASK [copr_project : Create Copr project] *********************************************************************************************************************************************************************************************
changed: [foreman-copr-with-el9]

TASK [copr_project : Configure chroots] ***********************************************************************************************************************************************************************************************
changed: [foreman-copr-with-el9] => (item={'name': 'rhel-8-x86_64', 'modules': ['ruby:2.7', 'python39:3.9', 'nodejs:14', 'postgresql:12'], 'external_repos': ['https://yum.puppet.com/puppet7/el/8/x86_64/', 'https://download.copr.fedorainfracloud.org/results/@theforeman/foreman-nightly-staging/rhel-8-x86_64'], 'comps_file': '/home/evgeni/Devel/theforeman/foreman-packaging/comps/comps-foreman-el8.xml', 'buildroot_packages': ['foreman-build']})
changed: [foreman-copr-with-el9] => (item={'name': 'rhel-9-x86_64', 'external_repos': ['https://yum.puppet.com/puppet7/el/9/x86_64/', 'https://download.copr.fedorainfracloud.org/results/@theforeman/foreman-nightly-staging/rhel-9-x86_64'], 'comps_file': '/home/evgeni/Devel/theforeman/foreman-packaging/comps/comps-foreman-el9.xml', 'buildroot_packages': []})

PLAY RECAP ****************************************************************************************************************************************************************************************************************************
foreman-copr-with-el9      : ok=3    changed=2    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0   
```

Try a release
```
% obal release rubygem-actioncable -e build_package_build_system=copr
…
TASK [build_package : Run build] ******************************************************************************************************************************************************************************************************
ok: [rubygem-actioncable] => (item=rhel-8-x86_64)
changed: [rubygem-actioncable] => (item=rhel-9-x86_64)

TASK [build_package : Created tasks] **************************************************************************************************************************************************************************************************
ok: [rubygem-actioncable] => 
  msg:
  - https://copr.fedorainfracloud.org/coprs/build/6711855
…
TASK [build_package : Watch copr build(s)] ********************************************************************************************************************************************************************************************
ok: [rubygem-actioncable]

PLAY RECAP ****************************************************************************************************************************************************************************************************************************
rubygem-actioncable        : ok=26   changed=3    unreachable=0    failed=0    skipped=21   rescued=0    ignored=0   

```

evgeni commented:
now running
```
% obal release rails_core_packages -e build_package_build_system=copr
```
